src/drivers/imu/adis16477/ADIS16477.cpp

```diff
@@ -255,79 +293,71 @@ ADIS16477::probe()
 {
 	DEVICE_DEBUG("probe");
 
-	reset();
-
 	// read product id (5 attempts)
 	for (int i = 0; i < 5; i++) {
+
+		if (reset() != PX4_OK) {
+			continue;
+		}
+
 		_product = read_reg16(PROD_ID);
 
 		if (_product == PROD_ID_ADIS16477) {
-			DEVICE_DEBUG("PRODUCT: %X", _product);
+			DEVICE_DEBUG("PRODUCT: %#X", _product);
 
-			if (self_test()) {
+			if (self_test_memory() && self_test_sensor()) {
```

Comment:
Any value is split and report what really failed?

Reply:
I don't think so, each will report the status register contents if anything fails.